feat: add spinner to subscribe block submit button

chickenn00dle opened the pull request:
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes https://app.asana.com/0/1207385073694408/1206756862605997/f

The subscribe block can take a few seconds to complete and the current loading state is not entirely clear for all publishers. This PR adds a loading spinner to the subscribe form submit button when clicked to make it more clear the form is loading:

https://github.com/Automattic/newspack-newsletters/assets/17905991/c18eda2d-89c5-4355-9c83-b8951dd48e9e

### How to test the changes in this Pull Request:

1. Verify reader activation is all set up on your test site
2. Add a newsletter subscription block to any page
3. Go to the relevant page as a reader
4. Input an email address and submit. Confirm a loading spinner appears within the submit button as shown above.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

chickenn00dle committed:
feat: add spinner to subscribe block submit button

laurelfulford reviewed (changes requested):
Ah, this is a nice addition to the block, @chickenn00dle! 🙌 

The only issue I ran into is when I change the button colour -- the styles "expect" the button to always be white text on red, so when it's changed, the text doesn't disappear, and the spinner doesn't match the color:

![image](https://github.com/Automattic/newspack-newsletters/assets/177561/c02e5a84-204a-42bc-a6f4-766fbbf17606)

The Newspack UI styles have an example of a spinner button that's not 100% functional (there's a style applied to a CSS class, but there isn't an example of the CSS class being applied/removed), but I think a couple things there could be used here for the colours issue:

* It has the button text [wrapped in a `span` and the span is given the style `visibility: hidden`](https://github.com/Automattic/newspack-plugin/blob/epic/ras-acc/assets/newspack-ui/scss/elements/forms/_buttons.scss#L200-L202) while loading, so it can't be seen, but it still maintains the size/width of the button.
* It uses [`currentcolor` and `transparency` for the spinner border colours](https://github.com/Automattic/newspack-plugin/blob/epic/ras-acc/assets/newspack-ui/scss/elements/forms/_buttons.scss#L207), so it picks up the text colour for the visible part, and is transparent for the invisible part.

Just let me know if you have any questions at all!

chickenn00dle committed:
fix: account for modified button colors

chickenn00dle commented:
Thank you @laurelfulford!

I updated the changes to use the same newspack-ui spinner strategy in https://github.com/Automattic/newspack-newsletters/pull/1529/commits/90906bb3a233182bb697b6f29054fc2f7f00f12f

I also updated a related PR in the main plugin (for registration form) to use the same strategy if you wanna tackle review for that one as well 🙏 : https://github.com/Automattic/newspack-plugin/pull/3180

laurelfulford reviewed (approved):
Thanks @chickenn00dle! That took care of it -- I tried with a few different colour combinations, and didn't see any further issues 🚢 

Edited to add: I'll try to get to the registration one today, too!
